junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/DisplayNameGenerator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api.extension;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.lang.reflect.Method;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.util.ClassUtils;
+
+/**
+ * {@code DisplayNameGenerator} defines the {@link Extension} API for
+ * generating display names programmatically.
+ *
+ * @since 5.4
+ */
+@API(status = EXPERIMENTAL, since = "5.4")
+public interface DisplayNameGenerator extends Extension {
+
+	default String generateDisplayNameForClass(Class<?> testClass) {
+		String name = testClass.getName();
+		int index = name.lastIndexOf('.');
+		return name.substring(index + 1);
+	}
+
+	default String generateDisplayNameForNestedClass(Class<?> nestedClass) {
```

Comment:
I have an issue with calling this "Nested".

I realize it's intended to only address `@Nested` test classes, but that is misleading at best, since a static nested test class (which is perfectly valid) is also a "NestedClass".

I'm thinking we should perhaps differentiate between static and non-static nested classes.

I have some other ideas about "static nested test classes" that I will likely address in a new issue.